store/sqlstore/channel_store.go

```diff
@@ -1814,6 +1815,70 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
+// CountUnreadSince gives the number of posts that a channel has since some date, this doesn't follow any threads.
+// this allows us to know how many posts have bean read already.
+func (s SqlChannelStore) CountUnreadSince(channelID string, since int64) (int64, *model.AppError) {
+	countUnreadQuery := `
+	SELECT count(*)
+	FROM Posts
+	WHERE
+		ChannelId = :channelId
+		AND CreateAt > :createAt
+		AND Type = ''
+		AND DeleteAt = 0
+	`
+	countParams := map[string]interface{}{
+		"channelId": channelID,
+		"createAt":  since,
+	}
+
+	unread, err := s.GetReplica().SelectInt(countUnreadQuery, countParams)
+	if err != nil {
+		return 0, model.NewAppError("SqlChannelStore.CountUnreadSince", "store.sql_channel.CountUnreadSince.app_error", countParams, fmt.Sprintf("Error while counting unreads for %s since %d", channelID, since), http.StatusInternalServerError)
```

Comment:
Can you include `err.Error()` somewhere in the fourth argument? If we're being consistent with other store methods, this would look more like
```suggestion
		return 0, model.NewAppError("SqlChannelStore.CountUnreadSince", "store.sql_channel.count_unread_since.app_error", nil, fmt.Sprintf("channel_id=%s, since=%d, err=%s", channelID, since, err), http.StatusInternalServerError)
```